src/convertToObject.js

```diff
@@ -6,7 +6,20 @@
  * @return {object}
  */
 function convertToObject(sourceString) {
-  // write your code here
+  const styles = sourceString.split(';');
+  const filteredStyles = styles.filter(
+    (styleDeclaration) => styleDeclaration.trim() !== '',
+  );
+
+  const styleObject = {};
+
+  filteredStyles.forEach((style) => {
```

Comment:
The use of `forEach` is acceptable here as it is an exception to the rule against modifying existing arrays/objects. However, consider using `reduce` for a more functional approach that aligns with the checklist's recommendation.